pkg/values/values.go

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package values
+
+import (
+	"reflect"
+	"strings"
+	"time"
+
+	"emperror.dev/errors"
+	"github.com/ghodss/yaml"
+	"github.com/mitchellh/mapstructure"
+)
+
+type Config map[string]interface{}
+
+func DecodeHook() mapstructure.DecodeHookFunc {
+	return func(a reflect.Type, b reflect.Type, d interface{}) (interface{}, error) {
+
+		// parse duration
+		if a.Kind() == reflect.String && b == reflect.TypeOf(new(time.Duration)).Elem() {
+			return time.ParseDuration(d.(string))
+		}
+
+		// parse values config
+		if a.Kind() == reflect.String && b == reflect.TypeOf(new(Config)).Elem() {
+
+			if data, ok := d.(string); ok {
+				output, err := toMap(data)
```

Comment:
I think this function could be inlined, we won't be reusing it any where and extracting it doesn't make to code easier to understand.